### PR TITLE
Fix for issue #230 - keep track of rows that had format conversion error when preparing for a bulk operation.

### PR DESCRIPTION
**Pull request by ashitsalesforce**

### CODE REVIEW

File: src/main/java/com/salesforce/dataloader/action/visitor/BulkLoadVisitor.java [hunk 1]
```diff
@@ -328,11 +327,28 @@ private void processResults(final DataReader dataReader, final BatchInfo batch,
...
+        // from the DataReader
+        while (rowCount < clientBatchInfo.numRows) {
+            boolean conversionSuccessOfRow = isRowConversionSuccessful(dataReaderRowNumBase + rowCount + additionalRowsToReadFromDataReader);
+            if (conversionSuccessOfRow) {
+                rowCount++;
+            } else {
+                additionalRowsToReadFromDataReader++;
+            }
+        }
+        totalRowsToReadFromDataReader += additionalRowsToReadFromDataReader;
```
Comment (jjangsfdc) [inline src/main/java/com/salesforce/dataloader/action/visitor/BulkLoadVisitor.java:346]: Might be easier to follow if it was initialized here
```suggestion
        final int totalRowsToReadFromDataReader = clientBatchInfo.numRows + additionalRowsToReadFromDataReader;
```

File: src/main/java/com/salesforce/dataloader/action/visitor/BulkQueryVisitor.java [hunk 2]
```diff
@@ -100,9 +104,33 @@ private void writeExtractionForBatch(BatchInfo batch) throws AsyncApiException,
 
         for (final String resultId : results.getResult()) {
             if (getProgressMonitor().isCanceled()) return;
+            boolean bufferResults = getController().getConfig().getBoolean(Config.BUFFER_UNPROCESSED_BULK_QUERY_RESULTS);
```
Comment (jjangsfdc) [inline src/main/java/com/salesforce/dataloader/action/visitor/BulkQueryVisitor.java:107]: Does this work?
```suggestion
            boolean bufferResults = getConfig().getBoolean(Config.BUFFER_UNPROCESSED_BULK_QUERY_RESULTS);
```

Comment (jjangsfdc) [reply src/main/java/com/salesforce/dataloader/action/visitor/BulkQueryVisitor.java:107]: Also, do you need to call it for every `for` iteration?

File: src/main/java/com/salesforce/dataloader/action/visitor/BulkLoadVisitor.java [hunk 3]
```diff
@@ -81,7 +81,6 @@
     private static final String ERROR_RESULT_COL = "Error";
     private static final String ID_RESULT_COL = "Id";
     private static final String CREATED_RESULT_COL = "Created";
-    private static final String SKIP_BATCH_ID = "SKIP";
```
[the remark this answers is not in the document]
Comment (ashitsalesforce) [reply src/main/java/com/salesforce/dataloader/action/visitor/BulkLoadVisitor.java:84]: The new mechanism to keep the track of each row to be processed on server side eliminates the need to have batches created in DL (not created by making a call to the server) with "SKIP" ID that were used to track the rows that failed local processing and format conversion. Consequently, the related constant is not used by the code anymore.

File: src/main/java/com/salesforce/dataloader/action/visitor/BulkLoadVisitor.java (hunk 1 above)
Comment (ashitsalesforce) [reply src/main/java/com/salesforce/dataloader/action/visitor/BulkLoadVisitor.java:346]: good point. I will make that change.

File: src/main/java/com/salesforce/dataloader/action/visitor/BulkQueryVisitor.java (hunk 2 above)
Comment (ashitsalesforce) [reply src/main/java/com/salesforce/dataloader/action/visitor/BulkQueryVisitor.java:107]: another good catch (2 of them). I will make the change.

**Pull request merged**